packages/flutter/lib/src/widgets/widget_span.dart

```diff
@@ -212,4 +244,167 @@ class WidgetSpan extends PlaceholderSpan {
+    renderObject
+      ..alignment = span.alignment
+      ..baseline = span.baseline
+      ..scale = textScaleFactor;
+  }
+}
+
+class _RenderScaledInlineWidget extends RenderBox with RenderObjectWithChildMixin<RenderBox> {
+  _RenderScaledInlineWidget(this._alignment, this._baseline, this._scale);
+
+  double get scale => _scale;
+  double _scale;
+  set scale(double value) {
+    if (value == _scale) {
+      return;
+    }
+    assert(value > 0);
+    assert(value.isFinite);
+    _scale = value;
+    markNeedsLayout();
+  }
+
+  ui.PlaceholderAlignment get alignment => _alignment;
+  ui.PlaceholderAlignment _alignment;
+  set alignment(ui.PlaceholderAlignment value) {
+    if (_alignment == value) {
+      return;
+    }
+    _alignment = value;
+    markNeedsLayout();
+  }
+
+  TextBaseline? get baseline => _baseline;
+  TextBaseline? _baseline;
+  set baseline(TextBaseline? value) {
+    if (value == _baseline) {
+      return;
+    }
+    _baseline = value;
+    markNeedsLayout();
+  }
+
+  @override
+  double computeMaxIntrinsicHeight(double width) {
+    return (child?.computeMaxIntrinsicHeight(width / scale) ?? 0.0) * scale;
+  }
+
+  @override
+  double computeMaxIntrinsicWidth(double height) {
+    return (child?.computeMaxIntrinsicWidth(height / scale) ?? 0.0) * scale;
+  }
+
+  @override
+  double computeMinIntrinsicHeight(double width) {
+    return (child?.computeMinIntrinsicHeight(width / scale) ?? 0.0) * scale;
+  }
+
+  @override
+  double computeMinIntrinsicWidth(double height) {
+    return (child?.computeMinIntrinsicWidth(height / scale) ?? 0.0) * scale;
+  }
+
+  @override
+  double? computeDistanceToActualBaseline(TextBaseline baseline) {
+    return switch (child?.getDistanceToActualBaseline(baseline)) {
+      null => super.computeDistanceToActualBaseline(baseline),
+      final double childBaseline => scale * childBaseline,
+    };
+  }
+
+  @override
+  Size computeDryLayout(BoxConstraints constraints) {
+    assert(!constraints.hasBoundedHeight);
+    final Size unscaledSize = child?.computeDryLayout(BoxConstraints(maxWidth: constraints.maxWidth / scale)) ?? Size.zero;
+    return unscaledSize * scale;
+  }
+
+  @override
+  void performLayout() {
+    final RenderBox? child = this.child;
+    if (child == null) {
+      return;
+    }
+    assert(!constraints.hasBoundedHeight);
+    // Only constrain the width to the maximum width of the paragraph.
+    // Leave height unconstrained, which will overflow if expanded past.
+    child.layout(BoxConstraints(maxWidth: constraints.maxWidth / scale), parentUsesSize: true);
+    size = child.size * scale;
+  }
+
+  @override
+  void applyPaintTransform(RenderBox child, Matrix4 transform) {
+    transform.scale(scale, scale);
+  }
+
+  @override
+  void paint(PaintingContext context, Offset offset) {
+    final RenderBox? child = this.child;
+    if (child == null) {
+      return;
+    }
+    context.pushTransform(
```

Comment:
Store the layerHandler this returns and pass it back in as oldLayer to give the engine more information for performance optimizations?

Reply:
Also, should we skip the transform if the scale is 1.0?